sklearn/neighbors/_binary_tree.pxi

```diff
@@ -511,24 +512,17 @@ cdef class NeighborsHeap:
     n_nbrs : int
         the size of each heap.
     """
-    cdef cnp.ndarray distances_arr
-    cdef cnp.ndarray indices_arr
-
     cdef DTYPE_t[:, ::1] distances
     cdef ITYPE_t[:, ::1] indices
 
     def __cinit__(self):
```

Comment:
```suggestion
    def __cinit__(self):
        # One-elements arrays are used as placeholders to prevent
        # any problem due to potential access to those attributes
        # (e.g. assigning to NULL or a to value in another segment).
```

Reply:
I wasn't aware of this trick, could you elaborate a bit more on this?